cmd/stdiscosrv/main.go

```diff
@@ -99,10 +99,14 @@ func main() {
 
 	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
 	if err != nil {
-		log.Println("Failed to load keypair. Generating one, this might take a while...")
-		cert, err = tlsutil.NewCertificate(certFile, keyFile, "stdiscosrv", 20*365)
-		if err != nil {
-			log.Fatalln("Failed to generate X509 key pair:", err)
+		if os.IsNotExist(err) {
+			log.Println("Failed to load keypair. Generating one, this might take a while...")
+			cert, err = tlsutil.NewCertificate(certFile, keyFile, "stdiscosrv", 20*365)
+			if err != nil {
+				log.Fatalln("Failed to generate X509 key pair:", err)
+			}
+		} else if err != nil {
```

Comment:
👍 now we don't need the outer `if err != nil` either

Reply:
That error is for if we have another error with regenrate a keypair

Reply:
Right. But `os.IsNotExist(nil)` is legal, and becomes false. So consider whether we need the extra `if err != nil` before it.